src/xmerl_dsig.erl

```diff
@@ -0,0 +1,730 @@
+            #'Certificate'{
+               tbsCertificate =
+                   #'TBSCertificate'{
+                      subjectPublicKeyInfo =
+                          #'SubjectPublicKeyInfo'{
+                             subjectPublicKey =
+                                 PubKey}}} = Cert,
+            KeyBin = case PubKey of
+                         {_, KeyBin2} -> KeyBin2;
+                         KeyBin3 -> KeyBin3
+                     end,
+            Key = public_key:pem_entry_decode({'RSAPublicKey', KeyBin,
+                                               not_encrypted}),
+            case public_key:verify(Data, HashFunction, Sig, Key) of
+                true ->
+                    case Fingerprints of
+                        any ->
+                            verify_signatures(Tail, Element, Fingerprints);
+                        [#'OTPCertificate'{}|_] ->
+                            logger:debug("Cert: ~p~n",
+                                      [public_key:pkix_decode_cert(CertBin, otp)]),
+                            case lists:any(
+                                   fun(CA) ->
+                                           public_key:pkix_is_issuer(
+                                             CertBin, CA)
+                                   end, Fingerprints) of
+                                true ->
+                                    verify_signatures(Tail, Element, Fingerprints);
+                                false ->
+                                    {error, cert_not_accepted}
+                            end;
+                        _ ->
+                            case lists:any(
+                                   fun(X) ->
+                                           lists:member(X, Fingerprints)
+                                   end, [CertHash, {sha,CertHash},
+                                         {sha256,CertHash2}]) of
+                                true ->
+                                    verify_signatures(Tail, Element, Fingerprints);
+                                false ->
+                                    {error, cert_not_accepted}
+                            end
+                    end;
+                false ->
+                    {error, bad_signature}
+            end
+    end;
+verify_signatures([], _, _) ->
+    ok.
+
+replicate_bugs(#xmlElement{nsinfo = {_, "deliverSecure"}} = Element) ->
+    logger:debug("replacing 'deliverSecure' with 'SealedDelivery'\n"),
+    Nss = Element#xmlElement.namespace#xmlNamespace.nodes,
+    {Prefix, _} = lists:keyfind('http://minameddelanden.gov.se/schema/Message/v3',
+                                2, Nss),
+    NewName = list_to_atom(Prefix ++ ":SealedDelivery"),
+    Element#xmlElement{ name = NewName
+                      , expanded_name = NewName
+                      , nsinfo = {Prefix, "SealedDelivery"}
+                      };
+replicate_bugs(Element) ->
+    Element.
```

Comment:
This looks like an application specific fixup filter that could just as well have been implemented in the code that uses this library. I think it should be extracted from here to there.

Reply:
resolved offline, this question was caused by my confusion from the Skatteverket references